### PR TITLE
Add localPV workload dashboard

### PR DESCRIPTION
**Pull request by rajaSahil**
Signed-off-by: Sahil Raja <sahil.raja@mayadata.io>

### What this PR does?

- Add dashboard to monitor localPV workload stats.
- Will add service and servicemonitor in the next PR.

#### NOTE
- These dashboards are built with the help of grafana variables.

#### To map persistentvolume_claim with localPV device using join, use this expression

- `kube_persistentvolumeclaim_info * on(volumename) group_left openebs_size_of_volume`

#### To map workload with localPV device using join, use this expression

- `openebs_size_of_volume * on(volumename) group_right(device) kube_persistentvolumeclaim_info * on(persistentvolumeclaim) group_left kube_pod_spec_volumes_persistentvolumeclaims_info * on(pod) group_left namespace_workload_pod:kube_pod_owner:relabel`

### How dashboards mapping are done?

- To map **workload** to **mountPoint** for filesystem dashboards:
  -  **Metrics hierarchy**
    Screenshot from 2021-06-28 09-33-43

- To map **workload** to **localPV device** for disk dashboards:
   
  * **Metrics hierarchy**
Screenshot from 2021-06-28 09-28-30

### Variable dependencies: 
  Screenshot from 2021-06-28 08-46-36

#### Screenshots 
Screenshot from 2021-06-14 16-19-36
Screenshot from 2021-06-12 08-30-20
Screenshot from 2021-06-12 08-12-27
Screenshot from 2021-06-12 08-12-36
Screenshot from 2021-06-12 08-12-49
Screenshot from 2021-06-12 08-12-59
Screenshot from 2021-06-12 08-13-09
Screenshot from 2021-06-12 08-13-21

**Note**
- This PR is dependent on lvm-metrics. PR: https://github.com/openebs/lvm-localpv/pull/94

**Ref**
- https://github.com/openebs/lvm-localpv/issues/78

### CODE REVIEW

File: deploy/charts/openebs-monitoring/values.yaml [hunk 1]
```diff
@@ -347,6 +362,53 @@ serviceMonitors:
...
+    ## Example
+
+    # selector:
+    #   matchLabels:
+    #     cas-type: localpv-lvm
+
+    selector:
+      matchLabels:
+        openebs.io/cas-type: localpv-lvm
+
```
Comment (Ab-hishek) [inline deploy/charts/openebs-monitoring/values.yaml]: Is this service present by default in the cluster for the lvm volumes? or the user has to first create the service?

Comment (rajaSahil) [reply deploy/charts/openebs-monitoring/values.yaml]: We will have to create service from lvm-localPV and add servicemonitor through here. 
I thought of doing this in this PR, but doing it in the next PR  will be good. Will raise the separate PR for adding service(lvm-localPV) as well as service monitor(openebs-monitoring).

**Comment by iyashu** [comment]
@kmova @rajaSahil Is it possible to directly import the dashboard json in grafana without doing helm install? I was trying to add above dashboard in my local cluster, but there seems to be some helm template vars being used in dashboard json as well.

**Comment by rajaSahil** [comment]
@iyashu There are no helm template vars used in the json. However, there were few links used that might have caused the error. I have removed those links. So can you please try again to import the updated json in grafana?

**Comment by iyashu** [comment]
Few improvements that we can do here - 

1. [P0] Having few set of metrics common across the localpv plugins will help in keeping all the dashboards, recording & alerts rules same regardless if one using the plugin or not. So in this case, a metric having a mapping of pv and device like `localpv_device{volumename="<pvc-name>", device="dm-0"} = 1` across both lvm & device localpv.

2. [P2] Additionally we can also see if we can have put information regarding pvc to node mapping (probably adding some label to pv or exposing additional metrics from controller plugin) without worrying about hosting node ready/accessible or not. It's not related to this pull request and we can take it up later on, but just putting it out here.
As of now we are using a kubelet_volume_stats_* metrics emitted by kubelet to figure out the all pvc hosted on a node, but if the node gets network partitioned or kubelet not running due to some reason, we'll not be able to distinguish two scenario where the node is network partitioned vs deleted from the cluster. Since network partitioned nodes may become ready again after some time, tenants may choose to avoid migrating their workloads.

**Review by avishnu — APPROVED** [review]
lgtm

**Review by Ab-hishek — APPROVED** [review]
lgtm

**Pull request merged**